harness/determined/common/experimental/resource_pool.py

```diff
@@ -107,3 +107,23 @@ def replace_bindings(
         bindings.put_OverwriteRPWorkspaceBindings(
             self._session, body=req, resourcePoolName=self.name
         )
+
+
+def _list(
+    session: api.Session,
+    offset: int = 0,
+    limit: int = 200,
+    pages: api.PageOpts = api.PageOpts.all,
+) -> List[bindings.v1ResourcePool]:
+    """List resource pools available to the cluster."""
+
+    def get_with_offset(inner_offset: int) -> bindings.v1GetResourcePoolsResponse:
+        return bindings.get_GetResourcePools(
+            session=session,
+            offset=inner_offset,
+            limit=limit,
+        )
+
+    resps = api.read_paginated(get_with_offset, offset, pages)
+
+    return [rp for r in resps if r.resourcePools is not None for rp in r.resourcePools]
```

Comment:
Even though they exist throughout the codebase, nested `for`s in list comprehensions are [not allowed per our style guide](https://google.github.io/styleguide/pyguide.html#274-decision). Please find another formulation. One choice is using a `for` loop, another is using `itertools` like:

```python
 itertools.chain.from_iterable(r.resourcePools for r in resps)
```

Reply:
I thought nested `for`s in list comprehension is preferred because it exists in various places in the codebase. I totally agree it's hard to read.